crates/kornia-core-ops/src/ops.rs

```diff
@@ -1,7 +1,76 @@
-use kornia_core::{storage::TensorStorage, SafeTensorType, Tensor, TensorAllocator};
+use kornia_core::{
+    storage::TensorStorage, tensor::get_strides_from_shape, CpuAllocator, SafeTensorType, Tensor,
+    TensorAllocator,
+};
 
 use crate::error::TensorOpsError;
 
+trait HasShape {
+    type WithShape<New: Shape>: HasShape<Shape = New>;
+    type Shape: Shape;
+}
+
+trait Shape {}
+
+trait ReduceShapeTo<Dst> {
+    fn reduce(self, dim: usize) -> Dst;
+}
+
+trait SumTo: Sized + HasShape {
+    fn sum<Dst: Shape>(&self, dim: usize) -> Result<Self::WithShape<Dst>, TensorOpsError>
+    where
+        Self::Shape: ReduceShapeTo<Dst>;
+}
+
+impl<T: SafeTensorType + std::ops::Add<Output = T>, A: TensorAllocator>
+    ReduceShapeTo<Tensor<T, 2, A>> for Tensor<T, 3, A>
+{
+    fn reduce(self, dim: usize) -> Tensor<T, 2, A> {
+        let shape = match dim {
+            0 => [self.shape[1], self.shape[2]],
```

Comment:
what if dim 0 is greater than 1 ? return error as it's not possible to perform the reduction ?